src/tracks/playabletrack/wavetrack/ui/WaveTrackAffordanceControls.cpp

```diff
@@ -0,0 +1,102 @@
+{
+    mFocusClip.reset();
+
+    std::vector<UIHandlePtr> results;
+
+    const auto track = FindTrack();
+
+    const auto waveTrack = std::static_pointer_cast<WaveTrack>(track->SubstitutePendingChangedTrack());
+
+    const auto rect = state.rect;
+
+    auto px = state.state.m_x;
+    auto py = state.state.m_y;
+
+    auto& trackPanel = TrackPanel::Get(*pProject);
+    auto& zoomInfo = *trackPanel.GetViewInfo();
+    for (const auto& clip : waveTrack->GetClips())
+    {
+        //TODO: we only need to calculate horizontal clip boundaries, but should be ok for demo
+        const ClipParameters params(false, waveTrack.get(), clip.get(), rect, SelectedRegion{}, zoomInfo);
+
+        if (px >= params.mid.GetLeft() && px <= params.mid.GetRight() &&
+            py >= params.mid.GetTop() && py <= params.mid.GetBottom())
+        {
+            results.push_back(AffordanceHandle::HitAnywhere(mAffordanceHandle, track));
+            mFocusClip = clip;
+            break;
+        }
+    }
+
+    return results;
+}
+
+void WaveTrackAffordanceControls::Draw(TrackPanelDrawingContext& context, const wxRect& rect, unsigned iPass)
+{
+    if (iPass == TrackArtist::PassBackground) {
+        auto track = FindTrack();
+        const auto artist = TrackArtist::Get(context);
+
+        TrackArt::DrawBackgroundWithSelection(context, rect, track.get(), artist->blankSelectedBrush, artist->blankBrush);
+
+        const auto waveTrack = std::static_pointer_cast<WaveTrack>(track->SubstitutePendingChangedTrack());
+        const auto& zoomInfo = *artist->pZoomInfo;
+
+        context.dc.SetClippingRegion(rect);
+
+        auto px = context.lastState.m_x;
+        auto py = context.lastState.m_y;
+
+        auto handle = mAffordanceHandle.lock();
+        for (const auto& clip : waveTrack->GetClips())
+        {
+            //TODO: we only need to calculate horizontal clip boundaries, but should be ok for demo
+            const ClipParameters params(false, waveTrack.get(), clip.get(), rect, SelectedRegion{}, zoomInfo);
+            wxRect affordanceRect{
+                params.mid.x - 1, //when rendering wave clip, 1px boundary is added by each side
+                params.mid.y,
+                params.mid.width + 2,
+                params.mid.height
+            };
+
+            auto selected = GetSelectedClip().lock() == clip;
+            auto highlight = selected ||
+                (px >= params.mid.GetLeft() && px <= params.mid.GetRight() &&
+                    py >= params.mid.GetTop() && py <= params.mid.GetBottom());
+            
+            TrackArt::DrawClipAffordance(context.dc, affordanceRect, highlight, selected);
+
+        }
+        context.dc.DestroyClippingRegion();
+    }
+}
+
+std::weak_ptr<WaveClip> WaveTrackAffordanceControls::GetSelectedClip() const
+{
+    if (auto handle = mAffordanceHandle.lock())
+    {
+        return handle->Clicked() ? mFocusClip : std::weak_ptr<WaveClip>();
```

Comment:
simplufy `: {}`

Reply:
do you mean use {} within ternary operator? I don't think C++ allows that

Reply:
I'm sure it does, at least in C++14 and later.  I have compile lines like it.

Reply:
None of my compilers allow that.
Clang bundled with xcode 12.4 says: Initializer list cannot be used on the right hand side of operator ':'
MSVC does not compile this as well. And minimum C++ Standard version is set to C++17 in CMake (i didn't change that)

Reply:
You are right, but nullptr will work instead of {}, and we should prefer that brevity and non-repetition of type information.

Reply:
Does not compile either: Non-pointer operand type 'const std::weak_ptr<WaveClip>' incompatible with nullptr

Reply:
oh well